### PR TITLE
A new class, Planetarium, to represent a system of spheres and a perspective

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

**Comment by pleroy** [comment]
retest this please

File: ksp_plugin/planetarium.hpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+using physics::RigidMotion;
+using physics::Trajectory;
+using quantities::Length;
+
+// A planetarium is a system of spheres together with a perspective.  In this
+// setting it is possible to draw trajectories in the projective plane.
+class Planetarium final {
+ public:
+  // TODO(phl): All this Navigation is weird.  Should it be named Plotting?
+  // In particular Navigration vs. NavigationFrame is a mess.
```
Comment (eggrobin) [inline ksp_plugin/planetarium.hpp]: especially if you misspell it.

File: ksp_plugin/planetarium.hpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  // TODO(phl): Maybe replace the spheres with an ephemeris.
```
Comment (eggrobin) [inline ksp_plugin/planetarium.hpp]: I think I'd like that, yes.

File: ksp_plugin_test/planetarium_test.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          Velocity<Barycentric>())));
+
+  Planetarium planetarium({sphere}, perspective, &plotting_frame);
+  auto const rp2_points =
+      planetarium.PlotMethod0(trajectory, Instant() + 10 * Second);
+
+  for (auto const& rp2_point : rp2_points) {
+    // The following limit is obtained by elementary geometry by noticing that
+    // the circle is viewed from the camera under an angle of π / 6.
+    EXPECT_LE(rp2_point.x(), 5.0 / Sqrt(3.0) * Metre);
```
Comment (eggrobin) [inline ksp_plugin_test/planetarium_test.cpp]: Use EXPECT_THAT and Le. also, why no upper bound?

File: ksp_plugin/planetarium.hpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      DiscreteTrajectory<Barycentric> const& trajectory,
+      Instant const& now) const;
+
+  // A naïve method that doesn't pay any attention to the perspective but tries
+  // to ensure that the points before the perspective are separated by less than
+  // |tolerance|.
+  std::vector<RP2Point<Length, Camera>> PlotMethod1(
+      Trajectory<Barycentric> const& trajectory,
+      Instant const& now,
+      Length const& tolerance) const;
```
Comment (eggrobin) [inline ksp_plugin/planetarium.hpp]: The PlotMethods should take a pair of iterators; most of the time we do not plot from `Begin` to `End`.

File: ksp_plugin/planetarium.hpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/planetarium.hpp]: Done.

File: ksp_plugin/planetarium.hpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/planetarium.hpp]: In a future PR [iraqiinformationminister].

File: ksp_plugin/planetarium.hpp (hunk 4 above)
Comment (pleroy) [inline ksp_plugin/planetarium.hpp]: Done.

File: ksp_plugin_test/planetarium_test.cpp (hunk 3 above)
Comment (pleroy) [inline ksp_plugin_test/planetarium_test.cpp]: Done.

**Pull request merged**